### PR TITLE
Do not return error if can't start smart update

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

File: pkg/controller/pxc/upgrade.go [hunk 1]
```diff
@@ -172,7 +172,8 @@ func (r *ReconcilePerconaXtraDBCluster) smartUpdate(sfs api.StatefulApp, cr *api
 	}
 
 	if sfs.StatefulSet().Status.ReadyReplicas < sfs.StatefulSet().Status.Replicas {
-		return fmt.Errorf("can't start/continue 'SmartUpdate': waiting for all replicas are ready")
+		log.Info("can't start/continue 'SmartUpdate': waiting for all replicas are ready")
```
Comment (delgod) [inline pkg/controller/pxc/upgrade.go:175]: should it be also `log.Info` if the backup is running in the line 170

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/pxc/upgrade.go (hunk 1 above)
Comment (blez) [reply pkg/controller/pxc/upgrade.go:175]: It can return real error witch should be logged as error, I think

Comment (delgod) [reply pkg/controller/pxc/upgrade.go:175]: it depends on error, just running backup is not the real error, it is just waiting till the moment when the backup is finished

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| storage | passed |
| scaling | passed |
| haproxy | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | passed |

File: pkg/controller/pxc/upgrade.go (hunk 1 above)
Comment (blez) [reply pkg/controller/pxc/upgrade.go:180]: https://github.com/percona/percona-xtradb-cluster-operator/pull/573/commits/3126ec3c6ea3e359a3c42b55104b7414f9b4e509

**Review by delgod — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | failed |

**Pull request merged**